test/extended/util/framework.go

```diff
@@ -0,0 +1,115 @@
+package util
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+
+	"github.com/openshift/origin/pkg/client"
+)
+
+// OsFramework provides helper functions and watchers for OpenShift
+type OsFramework struct {
+	Namespace *kapi.Namespace
+	Client    *client.Client
+}
+
+// NewOsFramework initialize OpenShift testing framework
+func NewOsFramework(ns *kapi.Namespace, c *client.Client) *OsFramework {
+	return &OsFramework{Namespace: ns, Client: c}
+}
+
+// WaitForABuild waits for a build to become Completed
+func (f *OsFramework) WaitForABuild(buildName string) error {
+	for {
```

Comment:
I don't see any timeout on this.  Is there some global timeout that will fail this eventually?